### PR TITLE
Bump minor to 1.3.0

### PR DESCRIPTION
**Pull request by jspaaks**
**Related issues**

Refs:

- N/A

**Describe the changes made in this pull request**

- This PR updates the `cff-version` from `1.2.0` to `1.3.0` in metadata, schema, schema guide, and relevant test files.
- I moved the isbn10 example from `1.2.0/pass/` to `1.2.0/fail/` because it is known to fail under `1.2.0` (So far it didn't show up because `main` is effectively at 1.2.1). 
- I copied `examples/1.2.0` to `examples/1.3.0` and updated the `cff-version` in the relevant files.
- I moved `examples/1.3.0/fail/isbn10` to `examples/1.3.0/pass` because under `1.3.0` it is expected to pass
- I moved the developer notes to a more visible location `README.dev.md`

**Review checklist**

- [x] Please check if the pull request is against the correct branch  
(format/schema/semantic documentation changes: `develop`; typos, meta files, etc.: `main`)
- [ ] Please check if all changes are recorded in `CHANGELOG.md` and adapt if necessary.
- [x] Please run tests locally.

```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout <branch>
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
<do other things>
```

- [ ] `<do other things>`

### CODE REVIEW

File: schema-guide.md [hunk 1]
```diff
@@ -4059,10 +4059,10 @@ When there are multiple licenses, it is assumed their relationship is OR, not AN
 - **description**: The version of a work.
 - **usage**:<br><br>
     ```yaml
-    version: "1.2.0"
+    version: "7.2.0"
```
Comment (jspaaks) [inline schema-guide.md:4062]: FYI I changed this to avoid confusion with `cff-version`

**Review by sdruskat — COMMENTED** [review]
Thanks @jspaaks 🍰!

Some comments:

- How did you update the versions? It didn't seem to have been through the `bump2version` tool that our project has installed and configured in `.bumpversion.cfg`? Related: #473.
- In addition to reviewing the file changes, I've `ack`ed `1\\.2\\.0` and all relevant instances seem to have been updated.

Other than that, this seems good to go 🥳!

**Comment by jspaaks** [comment]
> How did you update the versions? It didn't seem to have been through the bump2version tool that our project has installed and configured in .bumpversion.cfg? 

Indeed I didn't use bump2version -- I have trust issues :) -- I wouldn't mind removing it (refs #473)

**Comment by jspaaks** [comment]
Thanks for reviewing!

**Pull request merged**